src/trackinfoobject.cpp

```diff
@@ -738,23 +739,51 @@ QString TrackInfoObject::getURL() {
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveform() {
+    QMutexLocker lock(&m_qMutex);
     return m_waveform;
 }
 
 void TrackInfoObject::setWaveform(ConstWaveformPointer pWaveform) {
+    if (m_waveform == pWaveform) {
+        return;
+    }
+
+    QMutexLocker lock(&m_qMutex);
     m_waveform = pWaveform;
```

Comment:
Should we also fix the names of those members according to the naming conventions?

m_waveform(Summary) -> m_pWaveform(Summary)

The code looks a little odd with the parameter using the 'p' prefix, while the member does not.